sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/QuerySamples.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using static Azure.IoT.TimeSeriesInsights.Samples.SampleLogger;
+
+namespace Azure.IoT.TimeSeriesInsights.Samples
+{
+    internal class QuerySamples
+    {
+        /// <summary>
+        /// This sample demonstrates querying for raw events and aggregate series data from a Time Series Insights environment.
+        /// </summary>
+        public async Task RunSamplesAsync(TimeSeriesInsightsClient client, DeviceClient deviceClient)
+        {
+            // Figure out what keys make up the Time Series Id
+            TimeSeriesModelSettings modelSettings = await client.ModelSettings.GetAsync().ConfigureAwait(false);
+
+            int numOfIdKeys = modelSettings.TimeSeriesIdProperties.Count;
+
+            // Create a Time Series Id where the number of keys that make up the Time Series Id is fetched from Model Settings
+            var id = new List<string>();
+            for (int i = 0; i < numOfIdKeys; i++)
+            {
+                id.Add(Guid.NewGuid().ToString());
+            }
+
+            TimeSeriesId tsId = numOfIdKeys switch
+            {
+                1 => new TimeSeriesId(id[0]),
+                2 => new TimeSeriesId(id[0], id[1]),
+                3 => new TimeSeriesId(id[0], id[1], id[2]),
+                _ => throw new Exception($"Invalid number of Time Series Insights Id properties."),
+            };
+
+            // In order to query for data, let's first send events to the IoT Hub
+            await SendEventsToIotHubAsync(deviceClient, tsId, modelSettings.TimeSeriesIdProperties.ToArray()).ConfigureAwait(false);
+
+            // Sleeping for a few seconds to allow data to fully propagate in the Time Series Insights service
+            Thread.Sleep(TimeSpan.FromSeconds(3));
+
+            await RunQueryEventsSample(client, tsId).ConfigureAwait(false);
+
+            await RunQuerySeriesSample(client, tsId).ConfigureAwait(false);
+
+            await RunQueryAggregateSeriesSample(client, tsId).ConfigureAwait(false);
+        }
+
+        private async Task RunQueryAggregateSeriesSample(TimeSeriesInsightsClient client, TimeSeriesId tsId)
+        {
+            #region Snippet:TimeSeriesInsightsSampleQueryAggregateSeriesWithNumericVariable
+            Console.WriteLine("\n\nQuery for the average temperature over the past 30 seconds, in 2-second time slots.\n");
+
+            var numericVariable = new NumericVariable(
+                new TimeSeriesExpression("$event.Temperature"),
```

Comment:
It may be good idea to add a comment explaining how to build a time series expression? Like a link to msdocs which explain

Reply:
so instead of adding links to snippets, what are your thoughts on adding xml doc comments to this method (and the other similar methods) and add these links in the remarks section?

Reply:
I added a remarks section to the sample class.